modules/blazingsql/src/blazingcluster.ts

```diff
@@ -0,0 +1,320 @@
+
+export const CREATE_TABLE  = 'createTable';
+export const TABLE_CREATED = 'tableCreated';
+
+export const DROP_TABLE    = 'dropTable';
+export const TABLE_DROPPED = 'tableDropped';
+
+export const RUN_QUERY = 'runQuery';
+export const QUERY_RAN = 'ranQuery';
+
+export type ClusterProps = {
+  numWorkers: number,
+  ip: string,
+  port: number,
+};
+
+function _generateMessageId(ctxToken: number): string { return `message_${ctxToken}`; }
+
+let ctxToken = 0;
+
+export class BlazingCluster {
+  private workers: ChildProcess[];
+  // @ts-ignore
+  // instantiated within init()
+  private blazingContext: BlazingContext;
+
+  /**
+   * Initializes and returns an instance of BlazingCluster.
+   *
+   * @param clusterOptions optional options for the BlazingCluster instance
+   * @param contextConfigOptions optional options for the BlazingContext instance(s)
+   *
+   * @example
+   * ```typescript
+   * import {BlazingCluster} from '@rapidsai/blazingsql';
+   *
+   * await BlazingContext.init();
+   * ```
+   */
+  static async init(clusterOptions: Partial<ClusterProps>         = {},
+                    contextConfigOptions: Record<string, unknown> = {}): Promise<BlazingCluster> {
+    const {numWorkers = 1, ip = '0.0.0.0', port = 4000} = clusterOptions;
+
+    const bc            = new BlazingCluster(numWorkers);
+    const configOptions = {...defaultClusterConfigValues, ...contextConfigOptions};
+
+    const ucpMetadata = ['0', ...Object.keys(bc.workers)].map(
+      (_, idx) => { return ({workerId: idx.toString(), ip: ip, port: port + idx}); });
+
+    const createContextPromises: Promise<void>[] = [];
+    bc.workers.forEach((worker, idx) => {
+      createContextPromises.push(new Promise<void>((resolve) => {
+        const ralId = idx + 1;  // start ralId at 1 since ralId 0 is reserved for main process
+        worker.send({
+          operation: CREATE_BLAZING_CONTEXT,
+          ralId: ralId,
+          ucpMetadata: ucpMetadata,
+          configOptions: configOptions,
+          port: port,
+        });
+        worker.once('message', (msg: any) => {
+          const {operation}: {operation: string} = msg;
+          if (operation === BLAZING_CONTEXT_CREATED) { resolve(); }
+        });
+      }));
+    });
+
+    const ucpContext  = new UcpContext();
+    bc.blazingContext = new BlazingContext({
+      ralId: 0,
+      ralCommunicationPort: port,
+      configOptions: {...configOptions},
+      workersUcpInfo: ucpMetadata.map((xs) => ({...xs, ucpContext})),
+    });
+
+    await Promise.all(createContextPromises);
+
+    return bc;
+  }
+
+  private constructor(numWorkers: number) {
+    // If `__dirname` includes '/src' we are currently running a Jest test. Use a different relative
+    // path for when we are in a Jest test versus normal usage.
+    const relativePath = __dirname.includes('/src') ? './build/js/worker.js'
+                                                    : './modules/blazingsql/build/js/worker.js';
```

Comment:
We need to resolve this path from whatever `__dirname` is in this file. `modules/blazingsql` will not resolve to anything when this code is installed in a `node_modules` directory in the runtime containers.

Reply:
Should be fixed.